Add a page to preemptively answer assessors' questions

fureigh opened the pull request:
User stories (from initial conversation with @hollyallen):
 * As the head of infrastructure, I want plain-language talking points for explaining Cloud Foundry’s concepts and roles and how they relate to enforcing security on a development team in order to be able to explain the security implications to the assessor.
 * I want to be able to explain Cloud Foundry’s security implications to the assessor so that they feel reassured that cloud.gov meets the necessary security requirements. To that end, I want a plain-language overview of cloud.gov’s internal components and the security implications they have for development teams that host apps on cloud.gov.
 * I want plain language that explains the responsibilities of the platform vs the responsibilities of the customer so I can easily explain to the assessor the responsibility divide without referring to the customer responsibility matrix.
 * I want the assessor to be able to find the information they need directly on cloud.gov; one-on-one meetings don't scale.

For further context, see the relevant Favro cards:
 * https://favro.com/card/1e11108a2da81e3bd7153a7a/18F-7014
 * https://favro.com/card/1e11108a2da81e3bd7153a7a/18F-4993

fureigh committed:
Add a page to preemptively answer assessors' questions

Oh so drafty.

https://favro.com/card/1e11108a2da81e3bd7153a7a/18F-7014
https://favro.com/card/1e11108a2da81e3bd7153a7a/18F-4993

jcscottiii commented on content/overview/security/for-assessors.md:
```diff
@@ -0,0 +1,43 @@
+
+### Auditing login attempts
+
+Through multi-factor authentication and other means, the cloud.gov team verifies that people logging into cloud.gov legitimately own their accounts. But users don't log in to orgs and spaces, they log into cloud.gov itself. Accordingly, instead of monitoring cloud.gov logins/logouts, you'll want to audit what individual accounts did in the individual spaces to which they had access. For more info ...
```
at first glance, i'm not sure our tenants have access to the auditing of CC and UAA. would love to double check this.

mogul replied:
For the reason cited in the text, UAA auditing isn't really a thing that people should be looking at; the activity of logging in and out is "outside the boundary" of their orgs and spaces, if you know what I mean.

The doc says (emphasis mine) `**Operators** can use these logs...` I think that's it, they're not exposed to customers. We could probably do something to make them visible in an appropriate place, but there's no easy way to pull the relevant info out from the request data being logged currently... The log info includes the request and parameters, but we've found that there's no way to identify just from that which org or space might be affected. We've been talking to the CAPI team about this for a separate feature and it sounds like they might _want_ to provide a way to extract that information based on a request, but there's no way to do it yet.

mogul replied:
I think we have no recourse here other than to say that it's *possible* to have operators perform searches on those logs on behalf of customers, provided they can narrow the request to refer to specific users, ideally with some other fields fixed to narrow our search to find the relevant data. (cc @jmcarp since he brought up this feature in Slack)

fureigh replied:
Got it. Thanks for the clarification! I'll amend to something more general along the lines of @mogul's last comment.

fureigh replied:
Updated. @jcscottiii/@mogul, think it's worth including the following or is it just potentially confusing?

`For more information about which security events get logged and how, see https://docs.cloudfoundry.org/loggregator/cc-uaa-logging.html.`

fureigh committed:
Add link to app maintenance page

https://favro.com/card/1e11108a2da81e3bd7153a7a/18F-7014
https://favro.com/card/1e11108a2da81e3bd7153a7a/18F-4993

fureigh committed:
Update login auditing content

Incorporates feedback from https://github.com/18F/cg-site/pull/1017#pullrequestreview-56348115.

https://favro.com/card/1e11108a2da81e3bd7153a7a/18F-7014
https://favro.com/card/1e11108a2da81e3bd7153a7a/18F-4993

Commit:
Basic overview of how cloud.gov  works

Commit:
Merge branch 'master' into f-7014-add-material-targeting-assessors

Commit:
Merge remote-tracking branch 'origin/f-7014-add-material-targeting-assessors' into f-7014-add-material-targeting-assessors

Resolves merge conflicts
# Conflicts:
#	content/overview/security/for-assessors.md

fureigh commented on content/overview/security/for-assessors.md:
```diff
@@ -0,0 +1,49 @@
+
+1. Hardware infrastructure provided by Amazon Web Services's GovCloud environment including web servers, relational database servers, DNS routing services, and other systems the customer might need.
+1. A container providing the operating system and web server software (NGINX) and other basic software required to run any app. Apps are deployed into this container and isolated from other apps on the platform.
+1. Software components specific to the app the customer is launching are deployed along with the customer's app. Called buildpacks, these components provide the programming language and any dependencies required by the app. We strongly encourage customers to use the buildpacks we provide, though it is possible for customers to use their own.
```
@gboone Thanks for this! Let's restructure the final point for consistency with the others. Changing `launching are deployed` to `launching, deployed` should do it.

fureigh replied:
@gboone I'll make a commit and you can revert it if you dislike the phrasing.

fureigh committed:
Rephrase description of buildpacks

Per https://github.com/18F/cg-site/pull/1017/files#r134839844.

https://favro.com/card/1e11108a2da81e3bd7153a7a/18F-7014
https://favro.com/card/1e11108a2da81e3bd7153a7a/18F-4993

fureigh committed:
Remove TODOs

Move these to follow-up tasks.

https://favro.com/card/1e11108a2da81e3bd7153a7a/18F-7014
https://favro.com/card/1e11108a2da81e3bd7153a7a/18F-4993

fureigh committed:
Flesh out orgs, spaces, roles section for assessors

Drawing heavily on https://cloud.gov/docs/getting-started/concepts/,
also referencing http://docs.cloudfoundry.org/concepts/roles.html#roles.

https://favro.com/card/1e11108a2da81e3bd7153a7a/18F-7014
https://favro.com/card/1e11108a2da81e3bd7153a7a/18F-4993

fureigh committed:
Add Org Auditors to the list of roles

https://favro.com/card/1e11108a2da81e3bd7153a7a/18F-7014
https://favro.com/card/1e11108a2da81e3bd7153a7a/18F-4993

brittag commented on content/overview/security/for-assessors.md:
```diff
@@ -0,0 +1,59 @@
+
+## How cloud.gov works
+
+cloud.gov is a platform as a service. This means customers provide the code for their app and we provide the infrastructure, operating system, and supporting technologies required to run the app. There are three basic components to any cloud.gov app:
```
Do you mean "three basic components _underneath_ any cloud.gov app"?

fureigh replied:
@gboone ^

brittag commented on content/overview/security/for-assessors.md:
```diff
@@ -0,0 +1,59 @@
+
+cloud.gov is a platform as a service. This means customers provide the code for their app and we provide the infrastructure, operating system, and supporting technologies required to run the app. There are three basic components to any cloud.gov app:
+
+1. Hardware infrastructure provided by Amazon Web Services's GovCloud environment including web servers, relational database servers, DNS routing services, and other systems the customer might need.
```
AWS provides both hardware and software infrastructure - lots of software going on there, just infrastructure-level software instead of platform-level software.

brittag commented on content/overview/security/for-assessors.md:
```diff
@@ -0,0 +1,59 @@
+
+1. Hardware infrastructure provided by Amazon Web Services's GovCloud environment including web servers, relational database servers, DNS routing services, and other systems the customer might need.
+1. A container providing the operating system and web server software (NGINX) and other basic software required to run any app. Apps are deployed into this container and isolated from other apps on the platform.
+1. Software components specific to the app the customer is launching, deployed along with the customer's app. These components, called buildpacks, provide programming languages and other dependencies required by the app. We strongly encourage customers to use the buildpacks we provide, though it is possible for customers to use their own.
```
This may imply that the buildpacks contain all the dependencies that an app requires; many apps have [additional dependencies](https://github.com/18F/cg-site/blob/master/package.json)

fureigh replied:
Will clarify; thanks!

fureigh committed:
Revise explanations of security updates, what cloud.gov is

This commit seeks to address feedback from @brittag and @suprenant.

https://favro.com/card/1e11108a2da81e3bd7153a7a/18F-7014
https://favro.com/card/1e11108a2da81e3bd7153a7a/18F-4993

hollyallen commented on content/overview/security/for-assessors.md:
```diff
@@ -0,0 +1,61 @@
+
+1. Hardware and infrastructure-level software, including web servers, relational database servers, DNS routing services, and other systems the customer might need.
+1. A container providing the operating system and web server software (NGINX) and other basic software required to run any app. Apps are deployed into this container and isolated from other apps on the platform.
+1. Software components specific to the app the customer is launching, deployed along with the customer's app. These components, called buildpacks, provide programming languages and some additional dependencies required by the app. Though it's possible for customers to use their own buildpacks, we strongly encourage you to use the buildpacks we provide. That way you can benefit from our security u ...
```
"you" in this context is an assessor. Perhaps "we strongly encourage customers"?

hollyallen commented on content/overview/security/for-assessors.md:
```diff
@@ -0,0 +1,61 @@
+
+### Spaces
+
+Each org contains [spaces](http://docs.cloudfoundry.org/concepts/roles.html#spaces), which can contain applications. Applications in the same space share a location for app development, deployment, and maintenance.
```
Is it worth putting an example here, like "a customer may have spaces for production, staging, and development for example"?

hollyallen commented on content/overview/security/for-assessors.md:
```diff
@@ -0,0 +1,61 @@
+
+### Security updates
+
+When a new security update is available for one of cloud.gov's underlying technologies, we update the relevant buildpacks. These changes don't take effect on individual apps until the apps are restaged, though. You know your apps better than we do, and we prefer to minimize surprises, so your dev team is responsible for restaging your app(s). For more information, see the [app maintenance documen ...
```
Consider changing "you" to customers

fureigh committed:
Change `you` to `the customer`

This commit incorporates feedback from @hollyallen.

https://favro.com/card/1e11108a2da81e3bd7153a7a/18F-7014
https://favro.com/card/1e11108a2da81e3bd7153a7a/18F-4993

fureigh committed:
Elaborate on the explanation of apps in spaces

This commit incorporates feedback from @hollyallen.

https://favro.com/card/1e11108a2da81e3bd7153a7a/18F-7014
https://favro.com/card/1e11108a2da81e3bd7153a7a/18F-4993
